### PR TITLE
Additional Filters For Images & Extended Delays

### PR DESCRIPTION
**Pull request by Volnus**
This PR proposes a couple of new filters for getting images most notably those within text-widgets. In addition, it also extends the delay to the PHP_MAX and the reason for this is because many plugins and themes i found are (and I am guilty of this too in the past) of putting their delays as 9999999 which albeit is a bad practice but because of this, many themes (notably from a particular online marketplace) aren't getting their images lazy loaded. 

This change allows more images to be affected by the filter and pushes the delay to the max that it can be without us just slapping a bunch of 9's on the delay.

Fixes #

#### Changes proposed in this Pull Request:

*

#### Testing instructions:

*

### CODE REVIEW

File: modules/lazy-images/lazy-images.php [hunk 1]
```diff
@@ -48,15 +48,19 @@ private function __construct() {
 	}
 
 	public function setup_filters() {
-		add_filter( 'the_content', array( $this, 'add_image_placeholders' ), 99 ); // run this later, so other content filters have run, including image_add_wh on WP.com
-		add_filter( 'post_thumbnail_html', array( $this, 'add_image_placeholders' ), 11 );
-		add_filter( 'get_avatar', array( $this, 'add_image_placeholders' ), 11 );
+		add_filter( 'the_content', array( $this, 'add_image_placeholders' ), PHP_INT_MAX ); // run this later, so other content filters have run, including image_add_wh on WP.com
+		add_filter( 'post_thumbnail_html', array( $this, 'add_image_placeholders' ), PHP_INT_MAX );
+		add_filter( 'get_avatar', array( $this, 'add_image_placeholders' ), PHP_INT_MAX );
+		add_filter( 'widget_text', array($this, 'add_image_placeholders' ), PHP_INT_MAX );
```
Comment (ebinnion) [inline modules/lazy-images/lazy-images.php]: Nitpick, but could you please add a space between `array(` and `$this`.

File: modules/lazy-images/lazy-images.php [hunk 2]
```diff
@@ -48,15 +48,19 @@ private function __construct() {
...
+		add_filter( 'get_image_tag', array($this, 'add_image_placeholders' ), PHP_INT_MAX);
```
Comment (ebinnion) [inline modules/lazy-images/lazy-images.php]: Also, would you add the space here between `array(` and `$this` please.

File: modules/lazy-images/lazy-images.php [hunk 3]
```diff
@@ -48,15 +48,19 @@ private function __construct() {
...
 	}
 
 	public function remove_filters() {
-		remove_filter( 'the_content', array( $this, 'add_image_placeholders' ), 99 );
-		remove_filter( 'post_thumbnail_html', array( $this, 'add_image_placeholders' ), 11 );
-		remove_filter( 'get_avatar', array( $this, 'add_image_placeholders' ), 11 );
+		remove_filter( 'the_content', array( $this, 'add_image_placeholders' ), PHP_INT_MAX );
+		remove_filter( 'post_thumbnail_html', array( $this, 'add_image_placeholders' ), PHP_INT_MAX );
+		remove_filter( 'get_avatar', array( $this, 'add_image_placeholders' ), PHP_INT_MAX );
+		remove_filter( 'widget_text', array($this, 'add_image_placeholders' ), PHP_INT_MAX );
```
Comment (ebinnion) [inline modules/lazy-images/lazy-images.php]: Same with this line and the one below it.

**Comment by Volnus** [comment]
I fixed the spacing issues highlighted by @ebinnion

**Comment by oskosk** [comment]
@Volnus can you rebase this branch. There are some conflicts now. 

Thanks!


I'm relabeling as needs review.

**Pull request merged**